stage/stage.go

```diff
@@ -438,6 +446,11 @@ func (s *Stage) runQueries(ctx context.Context, queries []string, queryFile *str
 			}
 			log.Info().EmbedObject(result).Msgf("query finished")
 		}
+		// run post query cycle shell scripts
+		postQueryCycleErr := s.runShellScripts(ctx, s.PostQueryCycleShellScripts)
+		if retErr == nil {
+			retErr = postQueryCycleErr
```

Comment:
@ethanyzhang @agrawalreetika As we haven't tried it in real runs yet, this may or may not happen. But assume it happens because of some non-recoverable errors, the users won't be able to run any benchmark at all. Maybe we can consider issuing a warning but still continue the queries?